fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/AvailableFields.java

```diff
@@ -101,7 +101,7 @@ public static AvailableFields fromIndex(@Nonnull RecordType recordType,
 
         final List<KeyExpression> keyFields = new ArrayList<>();
         final List<KeyExpression> valueFields = new ArrayList<>();
-        if (indexTypes.getTextTypes().contains(index.getType())) {
+        if (indexTypes.getTextTypes().contains(index.getType()) || indexTypes.getLuceneTypes().contains(index.getType())) {
             // Full text index entries have all of their fields except the tokenized one.
```

Comment:
I'm not sure that's true. I think a Lucene text entry only has the `StoredField`'s, which is only the primary key. All the declared fields turn into indexed, but not stored, fields of the Lucene index. Unlike the existing text index, where the fields other than the tokenized ones are grouping fields.

Reply:
So I wasn't entirely sure of this. Both paths worked so I took a guess. Looking back through the code with a bit of a break this makes sense that the other path would be the one to take given that I don't think grouping fields are a part of this index. thanks.

Reply:
I think the answer to my previous question is _no_. Can we have a bunch of follow-up issues defined for this?